test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -1679,6 +1685,17 @@ set_reqs_based_on_capas(odp_tm_requirements_t *req)
 
 	if (tm_capabilities.tm_queue_shaper_supported)
 		req->tm_queue_shaper_needed = true;
+
+	/* Since packet priority mode effects only scheduler tests
+	 * and in our scheduler tests, we only test with set queues under
+	 * same parent node, we are fine with any packet priority mode.
+	 * This is because we are concerned with packet priority obeyed
+	 * at immediate parent node only.
+	 */
+	req->pkt_prio_mode = ODP_TM_PKT_PRIO_MODE_PRESERVE;
+	if (!tm_capabilities.pkt_prio_modes[ODP_TM_PKT_PRIO_MODE_PRESERVE])
+		req->pkt_prio_mode = ODP_TM_PKT_PRIO_MODE_OVERWRITE;
```

Comment:
Node priority is left unset. The code here enables the overwrite mode which requires that odp_node_param_t::priority is set to a valid value at TM node creation. The priority parameter is left unset (and is also not guaranteed to be initialized by odp_tm_node_params_init()).

So, create_tm_node() should initialize the priority parameter to a valid value. In addition to fixing the bug it makes validation test build fail if the parameter is entirely missing.